### ISSUE TITLE
 Incorrect file name length calculation

### ISSUE THREAD

**Issue by nick322** [mimo84/bootstrap-maxlength#146]
When I tested, I found that there was a calculation error in the file name of the uploaded file.
Here is the link I tested in jsfiddle
[jsfiddle link](https://jsfiddle.net/yj0m5r1z/)
![2020-02-21_141826](https://user-images.githubusercontent.com/9151347/75009439-a6a70980-54b5-11ea-9f0b-8a26e105058e.gif)

```
<input type="text" id="txt_report_title" name="report_title" maxlength="150" value="20200221_DRAMeXchange--Datacenter Demand Gradually Improves in 4Q19, with Three Giants to Enjoy Continual Growth of 8.8% in Server Memory Revenue.pdf" >
 <br> <br> <br> <br> <br> <br>
 <!--  filename = "20200221_DRAMeXchange--Datacenter Demand Gradually Improves in 4Q19, with Three Giants to Enjoy Continual Growth of 8.8% in Server Memory Revenue.pdf"-->
 <input type="file" id="file_report" name="report_title" maxlength="150"  >
```

```
$('input#txt_report_title').maxlength();
$('input#file_report').maxlength({
  alwaysShow: true
  
});
```

**Comment by djibe** [comment]
Hi, I don't understand.

*20200221_DRAMeXchange--Datacenter Demand Gradually Improves in 4Q19, with Three Giants to Enjoy Continual Growth of 8.8% in Server Memory Revenue.pdf*
is 149 characters long.

BS Maxlength counts 149 too.

Did I miss something ?

**Comment by nick322** [comment]
Hi djibe

Sorry I didn't describe it clearly enough.
I have attached a video file, I hope you understand.

I found that when I upload the file, the length of the file name will be calculated differently, it will be 161, not 149

[錄製_2020_02_22_01_58_43_824.zip](https://github.com/mimo84/bootstrap-maxlength/files/4237384/_2020_02_22_01_58_43_824.zip)

**Comment by djibe** [comment]
OK.

This is due to HTML DOM interaction with file input.

Open your fiddle or make a local file with JQuery.
Select your file.
Open Developer mode F12 > console
Send this code `$('#file_report').val();`
You get "C:\fakepath\20200221_DRAMeXchange--Datacenter Demand Gradually Improves in 4Q19, with Three Giants to Enjoy Continual Growth of 8.8% in Server Memory Revenue.pdf"

That is 161 characters isn't it ?

So, indeed, plugin needs a correction of calculation for file input.

Thanks for your report !

**Comment by nick322** [comment]
@djibe 

Haha, that's it. I didn't notice that "val ()" will show the full path.
And my user has no relevant webpage knowledge, so I don't know how to explain this situation.
I had to set (maxlength = "300") and skip this question first.

Thank you very much for your explanation.

**Comment by djibe** [comment]
Hi @nick322 .
This is in fact a "bug" or a feature.
I think you should reopen your issue until someone fixes it.

See you

**Comment by djibe** [comment]
Please close issue,
fixed in 1.9.0.
